docs/inflight_exit_scenarios.md

```diff
@@ -0,0 +1,409 @@
+# Simple In-flight Exit
+
+Alice sends tokens to Bob in transaction `tx` which has one input and one output. The block is withheld. 
+
+Bob calls `watcher/status` and gets a response:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "last_validated_child_block_number": 10000,
+        "last_mined_child_block_timestamp": 1535031020,
+        "last_mined_child_block_number": 11000,
+        "eth_syncing": true,
+        "byzantine_events": [
+            {
+                "event": "block_withholding",
+                "details": {
+                    "blockhash"  : "DB32876CC6F...",
+                    "blocknum"  : 10000,
+                }
+            }
+        ],
+        "inflight_txs": [
```

Comment:
might be worth pointing out that this is not in our immediate "features to implement" set. Until then, user only receives the `block_withholding` part and needs to know `txbytes` of in-flight txs themselves

Reply:
How will the user know which transactions are in-flight?
The sender may know that they've submitted a transaction that hasn't been put into a block, but the receiver probably won't.

Reply:
oh, but it's all about the sender, sender has initiative here. I've recently realized, that if the receiver starts the IFE, then the receiver is exposed to being slashed/grieved, [cf.](https://github.com/omisego/elixir-omg/blob/master/docs/morevp.md#honest-receiver-should-not-start-in-flight-exits)

Reply:
Yeah, agree with that scenario, but I don't see how the sender has an incentive to start an in-flight exit.

Ok, if there are multiple inputs from different address, then one of the senders _should_ start the in-flight, because then they have a chance of exiting that input if one of the _other_ senders is dishonest. A pretty rare edge case, IMO

However there is only one sender, they have nothing to gain by starting the exit.
If the sender is dishonest they can just start an MVP exit of the UTXO they spent in the in-flight transaction. Sure, this can be challenged by the receiver of the in-flight tx, but the not if the receiver doesn't know about it.
I guess someone else could challenge and the sender will lose their `standardExitBond`

But if the sender is honest, then they've 'spent' those tokens and it's not their problem if the receiver can't exit them. Unless they _want_ the receiver to exit in return for goods or something. Or maybe they have a significant amount of change in the transaction, then of course they will want to exit.

Hmm, maybe I've convinced myself :)

In any case, not implementing this yet affects some of the other api decisions and user flows e.g. I had assumed that the trigger to start an exit would be to see the user's address in the list.
But if we're not including this list it puts a lot more work on the user i.e. every time a user sends a transaction, they should store the details of it in case the chain goes byzantine and the transaction is in-flight. Are we comfortable with requiring wallets / apps to be responsible for this?

Reply:
if the tx is withheld or included in an invalid position and the sender doesn't open an IFE didn't send anything (e.g. goods shouldn't be dispatched). It should be seen as a failed transaction and sender is to blame here.

Oh wait, I just realized, that the sender cannot fully prove validity and "exitability" of the funds they send to the receiver :(. An inclusion proof isn't enough. So it seems I might be wrong about the "receiver shouldn't IFE" part or at least we need some additional protocol, which puts the full burden of staring IFEs on one of the parties, so that disputes can be resolved somehow.

My recent update to `morevp.md` might also be too optimistic then :(.

This way or another, in the first iteration, it is the user who should know their respective IFE's due. So in case the receiver needs that, they should receive the prompt out-of-band and track the IF-txs themselves. (note that the plan is to start handling that for the user in the immediate future)